libcontainer/container_linux.go

```diff
@@ -346,6 +346,20 @@ func (c *Container) start(process *Process) (retErr error) {
 	if err := utils.CloseExecFrom(3); err != nil {
 		return fmt.Errorf("unable to mark non-stdio fds as cloexec: %w", err)
 	}
+
+	// If the root in the container is different from current root user, we
+	// need to change the owner of stdio before we enter the user namespace,
+	// or else we may can't access stdio in the container.
+	// Please see https://github.com/opencontainers/runc/issues/4475
+	containerRootUID, err := c.config.HostRootUID()
```

Comment:
nit: this could just be called `rootUID` to avoid confusion between the terms "host" and "container".

Reply:
Actually, nevermind, let it stay as is.